src/mbgl/shader/circle_vertex.cpp

```diff
@@ -0,0 +1,13 @@
+#include <mbgl/shader/circle_vertex.hpp>
+#include <mbgl/shader/shader.hpp>
+#include <mbgl/gl/gl.hpp>
+
+namespace mbgl {
+
+void CircleVertex::bind(const int8_t* offset) {
+    static_assert(sizeof(CircleVertex) == 4, "expected CircleVertex size");
+
+    MBGL_BIND_VERTEX_ATTRIBUTE(CircleVertex, a_pos, offset);
```

Comment:
We still have the issue of Vertex types having to have the same names as the shaders. In our code, we (almost?) always do it like that, but there's no reason why we need to do it like that. I.e. `VertexType::a_pos` could be bound to `Shader::a_pos`, but we could also name the position in the `VertexType` something else.